# Analyzing UNICEF Global Education Data

In this notebook, I am using data provided by UNICEF to describe and map some key education metrics. I will provide supplementary analysis (maps, most likely) in R in the directory. I am particularly interested in visualizing these metrics to see global education disparities and perhaps see if any other indicators correlate with these education metrics. The raw dataset is located here, but I did some cleaning in Excel to make the parsing into Pandas easier: http://data.unicef.org/resources/state-worlds-children-2016-statistical-tables/

In [30]:
import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
from matplotlib import axes
import cartopy
import cartopy.io.shapereader as shpreader
import cartopy.crs as ccrs
from IPython.display import SVG, display

In [2]:
dataframe = pd.read_csv('SOWC-2016_all-tables_261.csv', na_values=['-'], skip_blank_lines=False, encoding = "ISO-8859-1")

In [3]:
dataframe

Country or Area Male Youth Literacy Rate  \
0                                  Afghanistan                       62   
1                                      Albania                       99   
2                                      Algeria                       94   
3                                      Andorra                        –   
4                                       Angola                       79   
5                          Antigua and Barbuda                        –   
6                                    Argentina                       99   
7                                      Armenia                      100   
8                                    Australia                        –   
9                                      Austria                        –   
10                                  Azerbaijan                      100   
11                                     Bahamas                        –   
12                                     Bahrain                       99   
13                                  Bangladesh                       79   
14                                    Barbados                        –   
15                                     Belarus                      100   
16                                     Belgium                        –   
17                                      Belize                        –   
18                                       Benin                       55   
19                                      Bhutan                       80   
20            Bolivia (Plurinational State of)                       99   
21                      Bosnia and Herzegovina                      100   
22                                    Botswana                       96   
23                                      Brazil                       97   
24                           Brunei Darussalam                       99   
25                                    Bulgaria                       98   
26                                Burkina Faso                       47   
27                                     Burundi                       90   
28                                  Cabo Verde                       98   
29                                    Cambodia                       88   
..                                         ...                      ...   
167                                   Suriname                       98   
168                                  Swaziland                       92   
169                                     Sweden                        –   
170                                Switzerland                        –   
171                       Syrian Arab Republic                       97   
172                                 Tajikistan                      100   
173                                   Thailand                       97   
174  The former Yugoslav Republic of Macedonia                       99   
175                                Timor-Leste                       80   
176                                       Togo                       87   
177                                      Tonga                       99   
178                        Trinidad and Tobago                      100   
179                                    Tunisia                       98   
180                                     Turkey                       99   
181                               Turkmenistan                      100   
182                                     Tuvalu                        –   
183                                     Uganda                       90   
184                                    Ukraine                      100   
185                       United Arab Emirates                       94   
186                             United Kingdom                        –   
187                United Republic of Tanzania                       76   
188                              United States          

Great, now we have our dataset loaded into pandas. We see there are a wide variety of indicators that we can look at from a global perspective. At the highest level, let's first calculate some basic statistics for each attribute. The data is stored as strings so I will have to cast them to numeric to run statistics. Note that there is missing data in this dataset, so the values may not be entirely accurate.

In [4]:
#make list of numeric columns for statistics
numeric_columns = []
for column in dataframe:
    if(column!='Country or Area'):
        numeric_columns.append(column)
print("{: <60} {: >10} {: >10}".format('Attribute', 'Mean', 'Median'))

for col in numeric_columns:
    mean = np.mean(pd.to_numeric(dataframe[col], errors='coerce'))
    median = np.nanmedian(pd.to_numeric(dataframe[col], errors='coerce'))
    print("{: <60} {: >10} {: >10}".format(col, str(round(mean,3)), str(median)))

Attribute                                                          Mean     Median
Male Youth Literacy Rate                                         89.846       97.0
Female Youth Literacy Rate                                       86.631       98.0
Mobile Phones (per 100)                                         103.531      106.5
Internet Users (per 100)                                         43.691       43.0
Male gross enrolment ratio pre-primary school                    62.793       70.5
Female gross enrolment ratio pre-primary school                   63.23       70.5
Male gross enrolment ratio primary school                       106.217      105.0
Female gross enrolment ratio primary school                     103.457      103.0
Male net enrolment ratio primary school                          90.764       95.0
Female net enrolment ratio primary school                        89.764       95.0
Male net attendance ratio primary school                         85.453       92.0
Fema

On their own, these global statistics do not mean much, but some interesting things do stand out. For example, the youth literacy rates are quite high! Additionally, I found it interesting that while globally there are about 103 mobile phones per 100 people, there are only around 43 internet users per 100 people. This shows an interesting discrepancy between the prevalence of mobile phones vs. a lack of internet access. Let's see which countries have the minimum and maximum value for each indicator to see if any trends emerge. Note that many countries or areas may have the same rates, but here only one is being selected.

In [5]:
print("{: <60} {: >20} {: >30}".format('Attribute', 'Max Country/Area', 'Min Country/Area'))
for col in numeric_columns:
    minCountry = dataframe.loc[pd.to_numeric(dataframe[col], errors='coerce').idxmin()]['Country or Area']
    maxCountry = dataframe.loc[pd.to_numeric(dataframe[col], errors='coerce').idxmax()]['Country or Area']
    #truncate names for printing 
    if(len(minCountry) > 20):
        minCountry = minCountry[0:20]
    
    if(len(maxCountry) > 20):
        maxCountry = maxCountry[0:20]
    print("{: <60} {: >20} {: >30}".format(col, maxCountry, minCountry))


Attribute                                                        Max Country/Area               Min Country/Area
Male Youth Literacy Rate                                                  Armenia                          Niger
Female Youth Literacy Rate                                                Armenia                          Niger
Mobile Phones (per 100)                                                    Kuwait                        Eritrea
Internet Users (per 100)                                                  Iceland           Democratic People's 
Male gross enrolment ratio pre-primary school                             Ecuador                           Chad
Female gross enrolment ratio pre-primary school                           Ecuador                           Chad
Male gross enrolment ratio primary school                                   Haiti                        Eritrea
Female gross enrolment ratio primary school                                 Haiti               

Let's plot the worst-performing countries in red and the best-performing countries in green to see the geospatial distribution. The R code used to develop these maps is in the directory as performance.Rmd. (Sidenote: I tried doing these in Python natively but R support for maps is just so much better!)

In [ ]:
#aggregate the best and worst-performing countries: the names were changed to ISO3 format for easy joining 
best_performing = ['ARM', 'KWT', 'ISL', 'ECU', 'HTI', 'CMR', 'HRV', 'NIU', 'JPN'
                   , 'ALB', 'QAT', 'BRB', 'PRK', 'BLR', 'ARG'] 
worst_performing = ['NER', 'ERI', 'PRK', 'TCD', 'LBR', 'BFA', 'PLW', 'UGA'
                   , 'COM', 'AGO', 'CAF', 'SOM']

#Construct dataframes from the lists so that we can write it to csv and pass to R for mapping - insert dummy valu
df1 = pd.DataFrame({'Country': best_performing, 'Value' : np.ones(len(best_performing))})
df2 = pd.DataFrame({'Country': worst_performing, 'Value' : np.ones(len(worst_performing))})
df1.to_csv('best_performing.csv')
df2.to_csv('worst_performing.csv')

As we can see in the attached maps, the worst-performing countries are all located in Africa, with the exception of North Korea. The best-performing countries are not bound by region. It may be enlightening to focus analysis on Africa to learn what factors can be improved and to see what additional data can be brought in to learn more about their effects on education outcomes. 